### ISSUE TITLE
Event coverage and priority

### ISSUE THREAD

**Issue by kmubiin** [kmubiin/by2020#10]
Raised from #1 and #4 with combined text from related comments.

To review internationalization (i18n) and localization (L10n) in this project, I have noted that I will make the information available in three languages only: English (default), Bahasa Malaysia, Japanese.

The event coverage and priority are related to the decision.

* Event coverage is **worldwide** or at least covering many countries, given that English is the default language in this project. That is, events are expected to be mostly discovered first in English, followed by Bahasa Malaysia and Japanese, either one or both that aware of the events.

* Event priority is **Malaysia**. That is, I will give priority to events in Malaysia first. I set the priority in this order: Malaysia, Japan, worldwide.

Just because the event coverage is worldwide, that does not mean this project will cover events in every country in the world. The intention is worldwide, or as far as English news would be disseminated and could reach to the knowledge of readers including myself.

As for the priority, I think that is more certain and probably unchanged.

My personal target is to commit 100 cumulative events in Malaysia, Japan and worldwide.

**Comment by kmubiin** [comment]
For quick review, event coverage is worldwide and events would be considered in this order of priority: Malaysia, Japan, worldwide. See examples for each category of event.

1. Malaysia:
  a. The 57th Malaysia Day (Hari Malaysia) [[1a]]
  b. The 63th National Day of Malaysia (Hari Merdeka) [[1b]]

2. Japan:
  a. The 2020 Summer Olympics (Tokyo 2020) [[2a]]
  b. Coming of Age Day [[2b]]

3. Worldwide:
  a. The 2020 World Expo (Expo Dubai 2020) [[3a]]
  b. The 55th National Day of Singapore [[3b]]
  c. World Water Day [[3c]]

These events have respective name, location and date and are scheduled to happen in 2020. Note that some events may have different date every year, such as 'Coming of Age Day' in Japan. For more events to observe, see [list of awareness day on Wikipedia][4].

I think the event coverage and priority is more or less clarified.


  [1a]: https://en.wikipedia.org/wiki/Malaysia_Day
  [1b]: https://en.wikipedia.org/wiki/Hari_Merdeka
  [2a]: https://en.wikipedia.org/wiki/2020_Summer_Olympics
  [2b]: https://en.wikipedia.org/wiki/Coming_of_Age_Day
  [3a]: https://en.wikipedia.org/wiki/Expo_2020
  [3b]: https://en.wikipedia.org/wiki/National_Day_(Singapore)
  [3c]: https://en.wikipedia.org/wiki/World_Water_Day
  [4]: https://en.wikipedia.org/wiki/List_of_awareness_days

**Comment by kmubiin** [comment]
In addition to what have been clarified above, this project may need consider events that occur naturally or unplanned. Example of such unplanned event:

* One-off kind of event such as opening ceremony, seasonal sale - probably better to ignore;
* Viral circulation of news via social media - consider the scale of influence and audience were being reached as a result, before proceeding to suggest;
* A natural disaster - depends on scale of damage and affected audience, as well as available slots to store the event, which means low priority.

Thing is, this project intends to store events that have been planned or scheduled, rather than events that occur naturally or unplanned. Priority will be given to the **planned** or **scheduled** events.

Unplanned events may be suggested, but will be processed only *after* all planned or scheduled events that have been submitted are considered. Such unplanned events may or may not be stored, depending on available capacity at the time.

This project is expected to handle up to 999 events per country, per year, per locale (according to the proposed design of Event Identification System). If total number of events has reached the limit, such unplanned events will not be stored.

**Comment by kmubiin** [comment]
One aspect of event coverage that may be confusing, is the entry number of event. In particular, the locale counterpart is to make the *same* event available in localized sources and text.

For example, `458.001-eng.json` is the JSON file for the first event in Malaysia in English text and `458.001-zsm.json` is the JSON file for the *same* event in Bahasa Melayu sources and text.

Should other locale accumulates events faster than English locale, then the entry number may be omitted for the corresponding events that are not available in English locale. Therefore, the events are accumulated in parallel; All locales has to follow the order of event.

* `458.001-eng.json` - Event 001 in English
* `458.001-jpn.json` - Event 001 in Japanese
* `458.001-zsm.json` - Event 001 in Bahasa Melayu
* `458.002-jpn.json` - Event 002 in Japanese
* `458.003-eng.json` - Event 003 in English
* `458.004-eng.json` - Event 004 in English
* `458.005-jpn.json` - Event 005 in Japanese
* `458.005-zsm.json` - Event 005 in Bahasa Melayu

In above list of JSON files, Event 001 and 005 have information for all three locales. However, Event 002 is only available in Japanese, and also Event 003 and 004 are only available in English.

That means, Event 002 was found at first from Japanese sources and not found in other locales. When a new event is found in English, the event will be stored as Event 003. The order of event discovery is respected, regardless of the locale.
